### PR TITLE
[Misc] Deprecation Warning when setting --engine-use-ray

### LINKED ISSUE #7045: [RFC]: Deprecation and removal for `--engine-use-ray`
### Motivation.

In the `async_engine` code path, we have an option to launch the engine in a separate process using Ray

```python
        parser.add_argument('--engine-use-ray',
                            action='store_true',
                            help='Use Ray to start the LLM engine in a '
                            'separate process as the server process
```

Originally, the option make it possible to separate the server's Python overhead with the engine's main scheduler loop. 

However, few factors made this unused/less popular
* Ray is an optional component, and typically not used in single node environment.
* The serialization and rpc typically offset the theoretical performance gain
* There are typically other ways to isolate server and engine (through multiprocessing, threading, etc).
* Recently, we are separating this in server using lower overhead approaches #6883

### Proposed Change.

Deprecation of the flag with warning for one release. 
Removal of the flag given no major pushbacks. 

### Feedback Period.

1wk

### CC List.

_No response_

### Any Other Things.

_No response_

### PR DESCRIPTION
**Pull request by wallashss**
Fix #7045 

This PR adds a deprecation warning when users set `--engine-use-ray`. 
For a concise message, I added in the message the issue number for additional clarification.

### CODE REVIEW

File: vllm/engine/async_llm_engine.py [hunk 1]
```diff
@@ -510,6 +511,12 @@ def __init__(self,
         self.log_requests = log_requests
         self.engine = self._init_engine(*args, **kwargs)
 
+        if self.engine_use_ray:
```
Comment (youkaichao) [inline vllm/engine/async_llm_engine.py:514]: we can hard fail here, and put an env var to turn it on  with a warning.

something like:

```python
if self.engine_use_ray:
    if envs.VLLM_ALLOW_ENGINE_USE_RAY:
        # print warning
    else:
        # directly error, with the error message pointing to the RFC issue
```

for the link, please use full url.

Comment (wallashss) [reply vllm/engine/async_llm_engine.py:514]: Thank you for your comment @youkaichao. I added modifications to address your suggestions.

**Review by youkaichao — APPROVED** [review]
LGTM, thanks for the contribution!

**Comment by njhill** [comment]
Thanks @wallashss!

Unfortunately some of the tests make use of this arg, we should update the tests to not use it.

If we want to get this PR merged in the meantime, we'll have to include changes to add the override env var to the tests in question.

**Comment by youkaichao** [comment]
we can run these tests with the override env var. we don't need to remove them right now.

**Comment by njhill** [comment]
> we can run these tests with the override env var. we don't need to remove them right now.

I think it is mostly tests for other things that make use of `--engine-use-ray` rather than tests for this option in itself, so we'll probably want to rework rather than remove them (agree that can be follow-on work though).

File: vllm/engine/arg_utils.py [hunk 2]
```diff
@@ -923,7 +923,13 @@ def add_cli_args(parser: FlexibleArgumentParser,
         parser.add_argument('--engine-use-ray',
                             action='store_true',
                             help='Use Ray to start the LLM engine in a '
-                            'separate process as the server process.')
+                            'separate process as the server process.'
+                            '(DEPRECATED. This argument is deprecated '
+                            'and may be removed in a future update. '
```
Comment (njhill) [inline vllm/engine/arg_utils.py:928]: suggest more decisive wording :)
```suggestion
                            'and will be removed in a future update. '
```

File: vllm/engine/async_llm_engine.py [hunk 3]
```diff
@@ -510,6 +511,20 @@ def __init__(self,
...
+            print_warning_once(
+                "DEPRECATED. `--engine-use-ray` is deprecated and may "
```
Comment (njhill) [inline vllm/engine/async_llm_engine.py:516]: same here
```suggestion
                "DEPRECATED. `--engine-use-ray` is deprecated and willl "
```

File: vllm/envs.py [hunk 4]
```diff
@@ -356,6 +357,13 @@ def get_default_config_root():
     lambda:
     (os.environ.get("VLLM_TEST_FORCE_FP8_MARLIN", "0").strip().lower() in
      ("1", "true")),
+
+    # If set, enable user to use ray which now is a deprecated feature
```
Comment (njhill) [inline vllm/envs.py:361]:
```suggestion
    # If set, allow running the engine as a separate ray actor, which is a deprecated feature soon to be removed
```

**Review by njhill — COMMENTED** [review]
Thanks @wallashss... just some minor change suggestions to the wording

**Comment by youkaichao** [comment]
hope @wallashss put the env var in all relevant places and the full test can pass 😉

**Comment by wallashss** [comment]
> hope @wallashss put the env var in all relevant places and the full test can pass 😉

I hope so too! I checked the tests and after several fixes, I did not find errors related to my changes. Right now I solved a conflict and a code format error. I guess that will be ready in the next round of checks.

**Comment by youkaichao** [comment]
tests look unrelated. merging.

**Comment by wallashss** [comment]
Thank you @youkaichao!

**Pull request merged**